burstphoto/exposure/exposure.swift

```diff
@@ -28,68 +28,77 @@ func correct_exposure(_ final_texture: MTLTexture, _ white_level: Int, _ black_l
             }
         }
         
-        var black_level0 = 0.0
-        var black_level1 = 0.0
-        var black_level2 = 0.0
-        var black_level3 = 0.0
-        
+        var black_levels_mean = Array(black_level[exp_idx].map{Double($0)})
+
         // if exposure levels are uniform, calculate mean value of all exposures
         if uniform_exposure {
-            
-            for comp_idx in 0..<exposure_bias.count {
-                black_level0 += Double(black_level[comp_idx][0])
-                black_level1 += Double(black_level[comp_idx][1])
-                black_level2 += Double(black_level[comp_idx][2])
-                black_level3 += Double(black_level[comp_idx][3])
+            for i in 0..<black_level.count {
```

Comment:
To make things a little bit more consistent, I would suggest to use "comp_idx" instead of "i" as the index. This fit better to the usage in other parts of the code. This would then allow for using the index "i" instead of "j" in the second loop one line below to stay consistent with the next loop performing the division a few lines below...

Reply:
I agree that I can use a better index name beyond the standard `i` and `j`, but I disagree that `comp_idx` is the one to use as I'm not comparing to anything. Take a look at the newest commit and let me know what you think, I've changed the names to something more descriptive (`img_idx` and `channel_idx`).